File: unique-paths/unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>>mat(m+1,vector<int>(n+1,0));
        for(int i=0;i<=m;i++){
            for(int j=0;j<=n;j++){
                if(i==1||j==1)
                    mat[i][j]=1;
                else
                    mat[i][j]=0;
            }
        }
        for(int i=2;i<=m;i++){
            for(int j=2;j<=n;j++){
                mat[i][j]+=mat[i-1][j]+mat[i][j-1];
            }
        }
        return mat[m][n];
    }
};

File: 62-unique-paths/unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        def solve(i,j,dp):
            if i >= n-1 or j >= m-1 :
                return 1
            if dp[i][j] != -1:
                return dp[i][j]
            dp[i][j] = solve(i+1,j,dp)  + solve(i,j+1,dp) 
            return dp[i][j]

        dp = [[-1] * (m+1) for _ in range(n+1)]
        return solve(0,0,dp)